contrib/neon/libpagestore.c

```diff
@@ -71,25 +71,6 @@ zenith_connect()
 				 errdetail_internal("%s", msg)));
 	}
 
-	/* Ask the Page Server to connect to us, and stream WAL from us. */
-	if (callmemaybe_connstring && callmemaybe_connstring[0]
```

Comment:
BTW it's possible to repurpose this code to help with old ps-1 version that requires callmemaybe. 

Compute can send callmemaybe to the pageserver with the following content:
```
callmemaybe tenant timeline any_sk_address
```

It will trick old pageserver into connecting to new safekeeper and can work until we implement a more complex migration.

Reply:
Yes, that's what I've thought was possible and tried to advertise that a few days ago, but so far the consensus was to move on and purge/bump ps-1 after we notify external users about this event.
This way, we don't have to add anything at all in the code 🙂

Reply:
@petuhovskiy AFAIK In current setup compute does not send callmemaybe connstr to pageserver. It is the leftover of old setup that allowed to run without safekeepers, so should be ok to remove that entirely.

Reply:
>  It is the leftover of old setup that allowed to run without safekeepers, so should be ok to remove that entirely.

Yeah, I'm talking about hacking it to work in a new way. 

>  but so far the consensus was to move on and purge/bump ps-1 after we notify external users about this event.

ps-1 on staging can be deleted, but on production we have to do migration somehow. Hacking callmemaybe in compute can be an alternative.

Reply:
We can don't do direct compute->pageserver START REPLICATION, and just ping pageserver with safekeeper address, so old pageserver will know one safekeeper where it needs to connect.

Reply:
ah, indeed